string_test.go

```diff
@@ -341,6 +341,20 @@ func TestStringAsNumber(t *testing.T) {
 		value2.chain.assertFailed(t)
 		num2.chain.assertFailed(t)
 		assert.Equal(t, float64(0), num2.Raw())
+
+		value3 := NewString(reporter, "8000000000000000")
+		num3 := value3.AsNumber(16)
+		value3.chain.assertOK(t)
+		num3.chain.assertOK(t)
+		assert.Equal(t, float64(9223372036854775808), num3.Raw())
```

Comment:
This test case may not be the expected output.
Converting `8000000000000000` from hexadecimal to decimal will be `9223372036854775808`,
but `num3.Raw()` is `9223372036854776000`.
This may be due to rounding by float64.

Reply:
Good catch!

I've pushed follow-up commit: 67cdcd4bfa69ab1c707405ad475e8175cd5eea2b

I've replaced 8000000000000000 with 4000000000000000 (2^62), the value from which float64 starts loosing precision for integers.

I also added precision loss checks to AsNumber, so that it fails if the string is an integer that can not be represented as float64 without loss, and added tests for that.

Reply:
I've created a new issue #191 for a more appropriate fix for this problem.